src/ast/implementation/declaration/error_definition.ts

```diff
@@ -0,0 +1,61 @@
+import { ASTNode } from "../../ast_node";
+import { ParameterList } from "../meta/parameter_list";
+import { StructuredDocumentation } from "../meta/structured_documentation";
+import { ContractDefinition } from "./contract_definition";
+
+export class ErrorDefinition extends ASTNode {
+    /**
+     * The name of the error
+     */
+    name: string;
+
+    /**
+     * The source range for name string
+     */
+    nameLocation?: string;
+
+    /**
+     * Optional documentation appearing above the error definition:
+     * - Is `undefined` when not specified.
+     * - Is type of `string` when specified and compiler version is older than `0.6.3`.
```

Comment:
There are no errors on compilers older tham 0.8.4 so should we allow `string` here?

Reply:
This is for compatibility reasons with legacy behavior. Sometimes it is just _convenient_ to pass string, especially if you do a lot of node creation.

I will fix the comment here, stating that is can be a string for compatibility reasons.